<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Draw Gene Arrow Maps in 'ggplot2' • gggenes</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Draw Gene Arrow Maps in 'ggplot2'">
<meta property="og:description" content="Provides a 'ggplot2' geom and helper functions for drawing gene
  arrow maps.">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">gggenes</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.4.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/introduction-to-gggenes.html">Introduction to 'gggenes'</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    

    
    

<div id="gggenes" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#gggenes" class="anchor"></a>gggenes</h1></div>
<p>gggenes is a <a href="https://ggplot2.tidyverse.org">ggplot2</a> extension for drawing gene arrow maps.</p>
<div id="installing-gggenes" class="section level2">
<h2 class="hasAnchor">
<a href="#installing-gggenes" class="anchor"></a>Installing gggenes</h2>
<p>Install the stable version of gggenes from CRAN:</p>
<p><code><a href="https://www.rdocumentation.org/packages/utils/topics/install.packages">install.packages("gggenes")</a></code></p>
<p>If you want the development version, install it from GitHub:</p>
<p><code><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">devtools::install_github("wilkox/gggenes")</a></code></p>
</div>
<div id="drawing-gene-arrows-with-geom_gene_arrow" class="section level2">
<h2 class="hasAnchor">
<a href="#drawing-gene-arrows-with-geom_gene_arrow" class="anchor"></a>Drawing gene arrows with <code><a href="reference/geom_gene_arrow.html">geom_gene_arrow()</a></code>
</h2>
<p><code><a href="reference/geom_gene_arrow.html">geom_gene_arrow()</a></code> is a ggplot2 geom that represents genes with arrows. The start and end locations of the genes within their molecule(s) are mapped to the <code>xmin</code> and <code>xmax</code> aesthetics respectively. These start and end locations are used to determine the directions in which the arrows point. The <code>y</code> aesthetic must be mapped to the molecule(s). If you are drawing more than one molecule, and the numerical locations of the genes are not similar across molecules, you almost certainly want to facet the plot with <code>scales = "free"</code> to avoid drawing ridiculously large molecules with ridiculously tiny genes.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(ggplot2)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(gggenes)</span>
<span id="cb1-3"><a href="#cb1-3"></a></span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(example_genes, <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">xmin =</span> start, <span class="dt">xmax =</span> end, <span class="dt">y =</span> molecule, <span class="dt">fill =</span> gene)) <span class="op">+</span></span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="st">  </span><span class="kw"><a href="reference/geom_gene_arrow.html">geom_gene_arrow</a></span>() <span class="op">+</span></span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span>(<span class="op">~</span><span class="st"> </span>molecule, <span class="dt">scales =</span> <span class="st">"free"</span>, <span class="dt">ncol =</span> <span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html">scale_fill_brewer</a></span>(<span class="dt">palette =</span> <span class="st">"Set3"</span>)</span></code></pre></div>
<p><img src="reference/figures/README-geom_gene_arrow-1.png" width="700"></p>
</div>
<div id="beautifying-the-plot-with-theme_genes" class="section level2">
<h2 class="hasAnchor">
<a href="#beautifying-the-plot-with-theme_genes" class="anchor"></a>Beautifying the plot with <code>theme_genes</code>
</h2>
<p>Because the resulting plot can look cluttered, a ggplot2 theme <code>theme_genes</code> is provided with some sensible defaults.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(example_genes, <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">xmin =</span> start, <span class="dt">xmax =</span> end, <span class="dt">y =</span> molecule, <span class="dt">fill =</span> gene)) <span class="op">+</span></span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="st">  </span><span class="kw"><a href="reference/geom_gene_arrow.html">geom_gene_arrow</a></span>() <span class="op">+</span></span>
<span id="cb2-3"><a href="#cb2-3"></a><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span>(<span class="op">~</span><span class="st"> </span>molecule, <span class="dt">scales =</span> <span class="st">"free"</span>, <span class="dt">ncol =</span> <span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html">scale_fill_brewer</a></span>(<span class="dt">palette =</span> <span class="st">"Set3"</span>) <span class="op">+</span></span>
<span id="cb2-5"><a href="#cb2-5"></a><span class="st">  </span><span class="kw"><a href="reference/theme_genes.html">theme_genes</a></span>()</span></code></pre></div>
<p><img src="reference/figures/README-theme_genes-1.png" width="700"></p>
</div>
<div id="aligning-genes-across-facets-with-make_alignment_dummies" class="section level2">
<h2 class="hasAnchor">
<a href="#aligning-genes-across-facets-with-make_alignment_dummies" class="anchor"></a>Aligning genes across facets with <code><a href="reference/make_alignment_dummies.html">make_alignment_dummies()</a></code>
</h2>
<p>Often you will want a certain gene to be vertically aligned across the faceted molecules. <code><a href="reference/make_alignment_dummies.html">make_alignment_dummies()</a></code> generates a set of dummy genes that if added to the plot with <code><a href="https://ggplot2.tidyverse.org/reference/geom_blank.html">geom_blank()</a></code> will extend the range of each facet to visually align the selected gene across facets.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>dummies &lt;-<span class="st"> </span><span class="kw"><a href="reference/make_alignment_dummies.html">make_alignment_dummies</a></span>(</span>
<span id="cb3-2"><a href="#cb3-2"></a>  example_genes,</span>
<span id="cb3-3"><a href="#cb3-3"></a>  <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">xmin =</span> start, <span class="dt">xmax =</span> end, <span class="dt">y =</span> molecule, <span class="dt">id =</span> gene),</span>
<span id="cb3-4"><a href="#cb3-4"></a>  <span class="dt">on =</span> <span class="st">"genE"</span></span>
<span id="cb3-5"><a href="#cb3-5"></a>)</span>
<span id="cb3-6"><a href="#cb3-6"></a></span>
<span id="cb3-7"><a href="#cb3-7"></a><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(example_genes, <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">xmin =</span> start, <span class="dt">xmax =</span> end, <span class="dt">y =</span> molecule, <span class="dt">fill =</span> gene)) <span class="op">+</span></span>
<span id="cb3-8"><a href="#cb3-8"></a><span class="st">  </span><span class="kw"><a href="reference/geom_gene_arrow.html">geom_gene_arrow</a></span>() <span class="op">+</span></span>
<span id="cb3-9"><a href="#cb3-9"></a><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_blank.html">geom_blank</a></span>(<span class="dt">data =</span> dummies) <span class="op">+</span></span>
<span id="cb3-10"><a href="#cb3-10"></a><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span>(<span class="op">~</span><span class="st"> </span>molecule, <span class="dt">scales =</span> <span class="st">"free"</span>, <span class="dt">ncol =</span> <span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb3-11"><a href="#cb3-11"></a><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html">scale_fill_brewer</a></span>(<span class="dt">palette =</span> <span class="st">"Set3"</span>) <span class="op">+</span></span>
<span id="cb3-12"><a href="#cb3-12"></a><span class="st">  </span><span class="kw"><a href="reference/theme_genes.html">theme_genes</a></span>()</span></code></pre></div>
<p><img src="reference/figures/README-make_alignment_dummies-1.png" width="700"></p>
</div>
<div id="labelling-genes-with-geom_gene_label" class="section level2">
<h2 class="hasAnchor">
<a href="#labelling-genes-with-geom_gene_label" class="anchor"></a>Labelling genes with <code><a href="reference/geom_gene_label.html">geom_gene_label()</a></code>
</h2>
<p>To label individual genes, provide a <code>label</code> aesthetic and use <code><a href="reference/geom_gene_label.html">geom_gene_label()</a></code>. <code><a href="reference/geom_gene_label.html">geom_gene_label()</a></code> uses the <a href="https://github.com/wilkox/ggfittext">ggfittext</a> package to fit the label text inside the gene arrows; see the ggfittext documentation for more details on how it resizes and reflows text to make it fit.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(</span>
<span id="cb4-2"><a href="#cb4-2"></a>    example_genes,</span>
<span id="cb4-3"><a href="#cb4-3"></a>    <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">xmin =</span> start, <span class="dt">xmax =</span> end, <span class="dt">y =</span> molecule, <span class="dt">fill =</span> gene, <span class="dt">label =</span> gene)</span>
<span id="cb4-4"><a href="#cb4-4"></a>  ) <span class="op">+</span></span>
<span id="cb4-5"><a href="#cb4-5"></a><span class="st">  </span><span class="kw"><a href="reference/geom_gene_arrow.html">geom_gene_arrow</a></span>(<span class="dt">arrowhead_height =</span> <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/reexports.html">unit</a></span>(<span class="dv">3</span>, <span class="st">"mm"</span>), <span class="dt">arrowhead_width =</span> <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/reexports.html">unit</a></span>(<span class="dv">1</span>, <span class="st">"mm"</span>)) <span class="op">+</span></span>
<span id="cb4-6"><a href="#cb4-6"></a><span class="st">  </span><span class="kw"><a href="reference/geom_gene_label.html">geom_gene_label</a></span>(<span class="dt">align =</span> <span class="st">"left"</span>) <span class="op">+</span></span>
<span id="cb4-7"><a href="#cb4-7"></a><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_blank.html">geom_blank</a></span>(<span class="dt">data =</span> dummies) <span class="op">+</span></span>
<span id="cb4-8"><a href="#cb4-8"></a><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span>(<span class="op">~</span><span class="st"> </span>molecule, <span class="dt">scales =</span> <span class="st">"free"</span>, <span class="dt">ncol =</span> <span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb4-9"><a href="#cb4-9"></a><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html">scale_fill_brewer</a></span>(<span class="dt">palette =</span> <span class="st">"Set3"</span>) <span class="op">+</span></span>
<span id="cb4-10"><a href="#cb4-10"></a><span class="st">  </span><span class="kw"><a href="reference/theme_genes.html">theme_genes</a></span>()</span></code></pre></div>
<p><img src="reference/figures/README-labelled_genes-1.png" width="700"></p>
</div>
<div id="reversing-some-genes-with-the-optional-forward-aesthetic" class="section level2">
<h2 class="hasAnchor">
<a href="#reversing-some-genes-with-the-optional-forward-aesthetic" class="anchor"></a>Reversing some genes with the optional <code>forward</code> aesthetic</h2>
<p>Sometimes you might want to reverse the direction of some genes from that implied by <code>xmin</code> and <code>xmax</code>. For example, you might want to draw both a forward and reverse strand within each facet, and reverse the direction of all the genes on the reverse strand. The optional <code>forward</code> aesthetic is intended for this sort of situation.</p>
<p>If <code>forward</code> is TRUE (the default), or any value that coerces to TRUE such as 1, the gene will be drawn pointing in the normal direction, i.e. that implied by <code>xmin</code> and <code>xmax</code>. If <code>forward</code> is FALSE, or any value that coerces to FALSE such as -1, the gene will be drawn in the reverse of this implied direction. In the following example, the <code>forward</code> aesthetic has been used to reverse the direction of all genes on the reverse strand from that implied by <code>xmin</code> and <code>xmax</code>.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>example_genes<span class="op">$</span>direction &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/ifelse">ifelse</a></span>(example_genes<span class="op">$</span>strand <span class="op">==</span><span class="st"> "forward"</span>, <span class="dv">1</span>, <span class="dv">-1</span>)</span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/subset">subset</a></span>(example_genes, molecule <span class="op">==</span><span class="st"> "Genome1"</span>),</span>
<span id="cb5-3"><a href="#cb5-3"></a>                <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">xmin =</span> start, <span class="dt">xmax =</span> end, <span class="dt">y =</span> strand, <span class="dt">fill =</span> gene,</span>
<span id="cb5-4"><a href="#cb5-4"></a>                    <span class="dt">forward =</span> direction)) <span class="op">+</span></span>
<span id="cb5-5"><a href="#cb5-5"></a><span class="st">  </span><span class="kw"><a href="reference/geom_gene_arrow.html">geom_gene_arrow</a></span>() <span class="op">+</span></span>
<span id="cb5-6"><a href="#cb5-6"></a><span class="st">  </span><span class="kw"><a href="reference/theme_genes.html">theme_genes</a></span>()</span></code></pre></div>
<p><img src="reference/figures/README-reversing_direction-1.png" width="700"></p>
</div>
<div id="viewing-subgene-segments" class="section level2">
<h2 class="hasAnchor">
<a href="#viewing-subgene-segments" class="anchor"></a>Viewing subgene segments</h2>
<p>We can highlight subgene segments, such as protein domains or local alignments, using <code><a href="reference/geom_subgene_arrow.html">geom_subgene_arrow()</a></code>.</p>
<p>This works similarly to <code><a href="reference/geom_gene_arrow.html">geom_gene_arrow()</a></code>, but in addition to <code>xmin</code> and <code>xmax</code> (which determine the gene boundaries), we need the aesthetics <code>xsubmin</code> and <code>xsubmax</code> to determine the subgene boundaries. <code><a href="reference/geom_gene_arrow.html">geom_gene_arrow()</a></code> will produce pretty arrowheads, as long as <code>xmin &gt;= xsubmin</code> and <code>xmax &gt;= xsubmax</code> for all subgenes (subgenes that break gene boundaries will be skipped with a warning).</p>
<p>The suggested usage is to use <code><a href="reference/geom_gene_arrow.html">geom_gene_arrow()</a></code> with no fill, and then add a subgene layer over this:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(example_genes, <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">xmin =</span> start, <span class="dt">xmax =</span> end, <span class="dt">y =</span> molecule)) <span class="op">+</span></span>
<span id="cb6-2"><a href="#cb6-2"></a><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span>(<span class="op">~</span><span class="st"> </span>molecule, <span class="dt">scales =</span> <span class="st">"free"</span>, <span class="dt">ncol =</span> <span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb6-3"><a href="#cb6-3"></a><span class="st">  </span><span class="kw"><a href="reference/geom_gene_arrow.html">geom_gene_arrow</a></span>(<span class="dt">fill =</span> <span class="st">"white"</span>) <span class="op">+</span></span>
<span id="cb6-4"><a href="#cb6-4"></a><span class="st">  </span><span class="kw"><a href="reference/geom_subgene_arrow.html">geom_subgene_arrow</a></span>(<span class="dt">data =</span> example_subgenes,</span>
<span id="cb6-5"><a href="#cb6-5"></a>    <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">xmin =</span> start, <span class="dt">xmax =</span> end, <span class="dt">y =</span> molecule, <span class="dt">fill =</span> gene,</span>
<span id="cb6-6"><a href="#cb6-6"></a>        <span class="dt">xsubmin =</span> from, <span class="dt">xsubmax =</span> to), <span class="dt">color=</span><span class="st">"black"</span>, <span class="dt">alpha=</span>.<span class="dv">7</span>) <span class="op">+</span></span>
<span id="cb6-7"><a href="#cb6-7"></a><span class="st">  </span><span class="kw"><a href="reference/theme_genes.html">theme_genes</a></span>()</span></code></pre></div>
<p><img src="reference/figures/README-subgenes-1.png" width="700"></p>
<p>To label subgenes, we can use <code><a href="reference/geom_subgene_label.html">geom_subgene_label()</a></code>, which works similarly to <code><a href="reference/geom_gene_label.html">geom_gene_label()</a></code> with the major difference that it requires <code>xsubmin</code> and <code>xsubmax</code> aesthetics (not <code>xmin</code> and <code>xmax</code>).</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/subset">subset</a></span>(example_genes, molecule <span class="op">==</span><span class="st"> "Genome4"</span> <span class="op">&amp;</span><span class="st"> </span>gene <span class="op">==</span><span class="st"> "genA"</span>),</span>
<span id="cb7-2"><a href="#cb7-2"></a>       <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">xmin =</span> start, <span class="dt">xmax =</span> end, <span class="dt">y =</span> strand)</span>
<span id="cb7-3"><a href="#cb7-3"></a>  ) <span class="op">+</span></span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="st">  </span><span class="kw"><a href="reference/geom_gene_arrow.html">geom_gene_arrow</a></span>() <span class="op">+</span></span>
<span id="cb7-5"><a href="#cb7-5"></a><span class="st">  </span><span class="kw"><a href="reference/geom_gene_label.html">geom_gene_label</a></span>(<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">label =</span> gene)) <span class="op">+</span></span>
<span id="cb7-6"><a href="#cb7-6"></a><span class="st">  </span><span class="kw"><a href="reference/geom_subgene_arrow.html">geom_subgene_arrow</a></span>(</span>
<span id="cb7-7"><a href="#cb7-7"></a>    <span class="dt">data =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/subset">subset</a></span>(example_subgenes, molecule <span class="op">==</span><span class="st"> "Genome4"</span> <span class="op">&amp;</span><span class="st"> </span>gene <span class="op">==</span><span class="st"> "genA"</span>),</span>
<span id="cb7-8"><a href="#cb7-8"></a>    <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">xsubmin =</span> from, <span class="dt">xsubmax =</span> to, <span class="dt">fill =</span> subgene)</span>
<span id="cb7-9"><a href="#cb7-9"></a>  ) <span class="op">+</span></span>
<span id="cb7-10"><a href="#cb7-10"></a><span class="st">  </span><span class="kw"><a href="reference/geom_subgene_label.html">geom_subgene_label</a></span>(</span>
<span id="cb7-11"><a href="#cb7-11"></a>    <span class="dt">data =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/subset">subset</a></span>(example_subgenes, molecule <span class="op">==</span><span class="st"> "Genome4"</span> <span class="op">&amp;</span><span class="st"> </span>gene <span class="op">==</span><span class="st"> "genA"</span>),</span>
<span id="cb7-12"><a href="#cb7-12"></a>    <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">xsubmin =</span> from, <span class="dt">xsubmax =</span> to, <span class="dt">label =</span> subgene),</span>
<span id="cb7-13"><a href="#cb7-13"></a>    <span class="dt">min.size =</span> <span class="dv">0</span></span>
<span id="cb7-14"><a href="#cb7-14"></a>  )</span></code></pre></div>
<p><img src="reference/figures/README-subgene_labels-1.png" width="700"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Download from CRAN at <br><a href="https://cloud.r-project.org/package=gggenes">https://​cloud.r-project.org/​package=gggenes</a>
</li>
<li>Report a bug at <br><a href="https://github.com/wilkox/gggenes/issues">https://​github.com/​wilkox/​gggenes/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>David Wilkins <br><small class="roles"> Author, maintainer </small>  </li>
<li><a href="authors.html">All authors...</a></li>
</ul>
</div>

      <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://travis-ci.org/wilkox/gggenes"><img src="https://travis-ci.org/wilkox/gggenes.svg?branch=master" alt="Travis-CI Build Status"></a></li>
<li><a href="https://cran.r-project.org/package=gggenes"><img src="http://www.r-pkg.org/badges/version/gggenes" alt="CRAN_Status_Badge"></a></li>
<li><a href="https://www.tidyverse.org/lifecycle/#maturing"><img src="https://img.shields.io/badge/lifecycle-maturing-blue.svg" alt="lifecycle"></a></li>
</ul>
</div>
</div>

</div>


      <footer><div class="copyright">
  <p>Developed by David Wilkins.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
