<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Introduction to 'gggenes' • gggenes</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to 'gggenes'">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">gggenes</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.6.0.9001</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fa-home fa-lg"></span></a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/introduction-to-gggenes.html">Introduction to 'gggenes'</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/wilkox/gggenes"><span class="fa fa-github"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Introduction to 'gggenes'</h1>
                        <h4 data-toc-skip class="author">David
Wilkins</h4>
            
            <h4 data-toc-skip class="date">2025-12-16</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/wilkox/gggenes/blob/master/vignettes/introduction-to-gggenes.Rmd" class="external-link"><code>vignettes/introduction-to-gggenes.Rmd</code></a></small>
      <div class="d-none name"><code>introduction-to-gggenes.Rmd</code></div>
    </div>

    
    
<p>‘gggenes’ is ggplot2 extension for drawing gene arrow maps.</p>
<div class="section level2">
<h2 id="drawing-gene-arrows-with-geom_gene_arrow">Drawing gene arrows with <code>geom_gene_arrow()</code><a class="anchor" aria-label="anchor" href="#drawing-gene-arrows-with-geom_gene_arrow"></a>
</h2>
<p><code><a href="../reference/geom_gene_arrow.html">geom_gene_arrow()</a></code> is a ‘ggplot2’ geom that represents
genes with arrows. The start and end locations of the genes within their
molecule(s) are mapped to the <code>xmin</code> and <code>xmax</code>
aesthetics respectively. These start and end locations are used to
determine the directions in which the arrows point. The <code>y</code>
aesthetic must be mapped to the molecule(s). If you are drawing more
than one molecule, and the numerical locations of the genes are not
similar across molecules, you almost certainly want to facet the plot
with <code>scales = "free"</code> to avoid drawing ridiculously large
molecules with ridiculously tiny genes.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkox.org/gggenes/">gggenes</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">example_genes</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>xmin <span class="op">=</span> <span class="va">start</span>, xmax <span class="op">=</span> <span class="va">end</span>, y <span class="op">=</span> <span class="va">molecule</span>, fill <span class="op">=</span> <span class="va">gene</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_gene_arrow.html">geom_gene_arrow</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">molecule</span>, scales <span class="op">=</span> <span class="st">"free"</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_fill_brewer</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set3"</span><span class="op">)</span></span></code></pre></div>
<p><img src="introduction-to-gggenes_files/figure-html/geom_gene_arrow-1.png" class="r-plt" width="576"></p>
</div>
<div class="section level2">
<h2 id="beautifying-the-plot-with-theme_genes">Beautifying the plot with <code>theme_genes()</code><a class="anchor" aria-label="anchor" href="#beautifying-the-plot-with-theme_genes"></a>
</h2>
<p>Because the resulting plot can look cluttered, a ‘ggplot2’ theme
<code><a href="../reference/theme_genes.html">theme_genes()</a></code> is provided with some sensible defaults.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">example_genes</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>xmin <span class="op">=</span> <span class="va">start</span>, xmax <span class="op">=</span> <span class="va">end</span>, y <span class="op">=</span> <span class="va">molecule</span>, fill <span class="op">=</span> <span class="va">gene</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_gene_arrow.html">geom_gene_arrow</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">molecule</span>, scales <span class="op">=</span> <span class="st">"free"</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_fill_brewer</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set3"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/theme_genes.html">theme_genes</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="introduction-to-gggenes_files/figure-html/theme_genes-1.png" class="r-plt" width="576"></p>
</div>
<div class="section level2">
<h2 id="aligning-genes-across-facets-with-make_alignment_dummies">Aligning genes across facets with
<code>make_alignment_dummies()</code><a class="anchor" aria-label="anchor" href="#aligning-genes-across-facets-with-make_alignment_dummies"></a>
</h2>
<p>Often you will want a certain gene to be vertically aligned across
the faceted molecules. <code><a href="../reference/make_alignment_dummies.html">make_alignment_dummies()</a></code> generates a
set of ‘dummy’ genes that if added to the plot with
<code><a href="https://ggplot2.tidyverse.org/reference/geom_blank.html" class="external-link">ggplot2::geom_blank()</a></code> will extend the range of each facet
to visually align the selected gene across facets.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dummies</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_alignment_dummies.html">make_alignment_dummies</a></span><span class="op">(</span></span>
<span>  <span class="va">example_genes</span>,</span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>xmin <span class="op">=</span> <span class="va">start</span>, xmax <span class="op">=</span> <span class="va">end</span>, y <span class="op">=</span> <span class="va">molecule</span>, id <span class="op">=</span> <span class="va">gene</span><span class="op">)</span>,</span>
<span>  on <span class="op">=</span> <span class="st">"genE"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">example_genes</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>xmin <span class="op">=</span> <span class="va">start</span>, xmax <span class="op">=</span> <span class="va">end</span>, y <span class="op">=</span> <span class="va">molecule</span>, fill <span class="op">=</span> <span class="va">gene</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_gene_arrow.html">geom_gene_arrow</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_blank.html" class="external-link">geom_blank</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">dummies</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">molecule</span>, scales <span class="op">=</span> <span class="st">"free"</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_fill_brewer</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set3"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/theme_genes.html">theme_genes</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="introduction-to-gggenes_files/figure-html/make_alignment_dummies-1.png" class="r-plt" width="576"></p>
</div>
<div class="section level2">
<h2 id="labelling-genes-with-geom_gene_label">Labelling genes with <code>geom_gene_label()</code><a class="anchor" aria-label="anchor" href="#labelling-genes-with-geom_gene_label"></a>
</h2>
<p>To label individual genes, provide a <code>label</code> aesthetic and
use <code><a href="../reference/geom_gene_label.html">geom_gene_label()</a></code>. <code><a href="../reference/geom_gene_label.html">geom_gene_label()</a></code> uses
the <a href="https://github.com/wilkox/ggfittext" class="external-link">‘ggfittext’</a>
package to fit the label text inside the gene arrows; see the
‘ggfittext’ documentation for more details on how it resizes and reflows
text to make it fit.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">example_genes</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>xmin <span class="op">=</span> <span class="va">start</span>, xmax <span class="op">=</span> <span class="va">end</span>, y <span class="op">=</span></span>
<span>                                            <span class="va">molecule</span>, fill <span class="op">=</span> <span class="va">gene</span>, label <span class="op">=</span> <span class="va">gene</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_gene_arrow.html">geom_gene_arrow</a></span><span class="op">(</span>arrowhead_height <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">3</span>, <span class="st">"mm"</span><span class="op">)</span>, arrowhead_width <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">1</span>, <span class="st">"mm"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_gene_label.html">geom_gene_label</a></span><span class="op">(</span>align <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_blank.html" class="external-link">geom_blank</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">dummies</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">molecule</span>, scales <span class="op">=</span> <span class="st">"free"</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_fill_brewer</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set3"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/theme_genes.html">theme_genes</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="introduction-to-gggenes_files/figure-html/labelled_genes-1.png" class="r-plt" width="576"></p>
</div>
<div class="section level2">
<h2 id="reversing-gene-direction-with-the-optional-forward-aesthetic">Reversing gene direction with the optional <code>forward</code>
aesthetic<a class="anchor" aria-label="anchor" href="#reversing-gene-direction-with-the-optional-forward-aesthetic"></a>
</h2>
<p>By default, genes arrows are drawn pointing in the direction implied
by <code>xmin</code> and <code>xmax</code>. You can override this with
the optional <code>forward</code> aesthetic, which can be useful when
the gene coordinates and orientation are encoded as separate
variables.</p>
<p>If <code>forward</code> is TRUE (the default), or any value that
coerces to TRUE such as 1, the gene will be drawn pointing in the
implied direction, i.e. from <code>xmin</code> to <code>xmax</code>. If
<code>forward</code> is FALSE, or any value that coerces to FALSE such
as 0, the gene will be drawn in the reverse of this implied
direction:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">example_genes</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>xmin <span class="op">=</span> <span class="va">start</span>, xmax <span class="op">=</span> <span class="va">end</span>, y <span class="op">=</span> <span class="va">molecule</span>, fill <span class="op">=</span> <span class="va">gene</span>, </span>
<span>                          forward <span class="op">=</span> <span class="va">orientation</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_gene_arrow.html">geom_gene_arrow</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">molecule</span>, scales <span class="op">=</span> <span class="st">"free"</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_fill_brewer</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set3"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/theme_genes.html">theme_genes</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="introduction-to-gggenes_files/figure-html/reversing_direction-1.png" class="r-plt" width="576"></p>
</div>
<div class="section level2">
<h2 id="viewing-subgene-segments">Viewing subgene segments<a class="anchor" aria-label="anchor" href="#viewing-subgene-segments"></a>
</h2>
<p>We can highlight subgene segments, such as protein domains or local
alignments, using <code><a href="../reference/geom_subgene_arrow.html">geom_subgene_arrow()</a></code>.</p>
<p>This works similarly to <code><a href="../reference/geom_gene_arrow.html">geom_gene_arrow()</a></code>, but in
addition to <code>xmin</code> and <code>xmax</code> (which determine the
gene boundaries), we need the aesthetics <code>xsubmin</code> and
<code>xsubmax</code> to determine the subgene boundaries.
<code><a href="../reference/geom_gene_arrow.html">geom_gene_arrow()</a></code> will produce pretty arrowheads, as long
as <code>xmin &gt;= xsubmin</code> and <code>xmax &gt;= xsubmax</code>
for all subgenes (subgenes that break gene boundaries will be skipped
with a warning).</p>
<p>The suggested usage is to use <code><a href="../reference/geom_gene_arrow.html">geom_gene_arrow()</a></code> with no
fill, and then add a subgene layer over this:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">example_genes</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>xmin <span class="op">=</span> <span class="va">start</span>, xmax <span class="op">=</span> <span class="va">end</span>, y <span class="op">=</span> <span class="va">molecule</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">molecule</span>, scales <span class="op">=</span> <span class="st">"free"</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_gene_arrow.html">geom_gene_arrow</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_subgene_arrow.html">geom_subgene_arrow</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">example_subgenes</span>,</span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>xmin <span class="op">=</span> <span class="va">start</span>, xmax <span class="op">=</span> <span class="va">end</span>, y <span class="op">=</span> <span class="va">molecule</span>, fill <span class="op">=</span> <span class="va">gene</span>,</span>
<span>        xsubmin <span class="op">=</span> <span class="va">from</span>, xsubmax <span class="op">=</span> <span class="va">to</span><span class="op">)</span>, color<span class="op">=</span><span class="st">"black"</span>, alpha<span class="op">=</span><span class="fl">.7</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/theme_genes.html">theme_genes</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="introduction-to-gggenes_files/figure-html/subgenes-1.png" class="r-plt" width="576"></p>
<p>To label subgenes, we can use <code><a href="../reference/geom_subgene_label.html">geom_subgene_label()</a></code>,
which works similarly to <code><a href="../reference/geom_gene_label.html">geom_gene_label()</a></code> with the major
difference that it requires <code>xsubmin</code> and
<code>xsubmax</code> aesthetics (not <code>xmin</code> and
<code>xmax</code>).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">example_genes</span>, <span class="va">molecule</span> <span class="op">==</span> <span class="st">"Genome4"</span> <span class="op">&amp;</span> <span class="va">gene</span> <span class="op">==</span> <span class="st">"genA"</span><span class="op">)</span>,</span>
<span>       <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>xmin <span class="op">=</span> <span class="va">start</span>, xmax <span class="op">=</span> <span class="va">end</span>, y <span class="op">=</span> <span class="va">strand</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_gene_arrow.html">geom_gene_arrow</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_gene_label.html">geom_gene_label</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label <span class="op">=</span> <span class="va">gene</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_subgene_arrow.html">geom_subgene_arrow</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">example_subgenes</span>, <span class="va">molecule</span> <span class="op">==</span> <span class="st">"Genome4"</span> <span class="op">&amp;</span> <span class="va">gene</span> <span class="op">==</span> <span class="st">"genA"</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>xsubmin <span class="op">=</span> <span class="va">from</span>, xsubmax <span class="op">=</span> <span class="va">to</span>, fill <span class="op">=</span> <span class="va">subgene</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_subgene_label.html">geom_subgene_label</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">example_subgenes</span>, <span class="va">molecule</span> <span class="op">==</span> <span class="st">"Genome4"</span> <span class="op">&amp;</span> <span class="va">gene</span> <span class="op">==</span> <span class="st">"genA"</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>xsubmin <span class="op">=</span> <span class="va">from</span>, xsubmax <span class="op">=</span> <span class="va">to</span>, label <span class="op">=</span> <span class="va">subgene</span><span class="op">)</span>,</span>
<span>    min.size <span class="op">=</span> <span class="fl">0</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p><img src="introduction-to-gggenes_files/figure-html/subgene%20labels-1.png" class="r-plt" width="576"></p>
</div>
<div class="section level2">
<h2 id="drawing-point-genetic-features">Drawing point genetic features<a class="anchor" aria-label="anchor" href="#drawing-point-genetic-features"></a>
</h2>
<p>We can draw point genetic features, such as restriction sites or
transcription start sites, with <code><a href="../reference/geom_feature.html">geom_feature()</a></code>, and label
them with <code><a href="../reference/geom_feature_label.html">geom_feature_label()</a></code>. Both of these geoms take an
optional <code>forward</code> aesthetic to indicate whether a feature is
oriented and, if so, in what direction (<code>TRUE</code> meaning
oriented towards the right and <code>FALSE</code> meaning towards the
left).</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">example_genes</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>xmin <span class="op">=</span> <span class="va">start</span>, xmax <span class="op">=</span> <span class="va">end</span>, y <span class="op">=</span> <span class="va">molecule</span>, fill <span class="op">=</span> <span class="va">gene</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_feature.html">geom_feature</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">example_features</span>,</span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">position</span>, y <span class="op">=</span> <span class="va">molecule</span>, forward <span class="op">=</span> <span class="va">forward</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_feature_label.html">geom_feature_label</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">example_features</span>,</span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">position</span>, y <span class="op">=</span> <span class="va">molecule</span>, label <span class="op">=</span> <span class="va">name</span>, forward <span class="op">=</span> <span class="va">forward</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_gene_arrow.html">geom_gene_arrow</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_blank.html" class="external-link">geom_blank</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">example_dummies</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">molecule</span>, scales <span class="op">=</span> <span class="st">"free"</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_fill_brewer</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set3"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/theme_genes.html">theme_genes</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="introduction-to-gggenes_files/figure-html/unnamed-chunk-2-1.png" class="r-plt" width="576"></p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by David Wilkins.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
